File: src/org/cappuccino_project/ide/intellij/plugin/indices/ObjJUnifiedMethodIndex.java
package org.cappuccino_project.ide.intellij.plugin.indices;

import com.intellij.psi.stubs.StubIndexKey;
import org.cappuccino_project.ide.intellij.plugin.psi.interfaces.ObjJMethodHeaderDeclaration;
import org.jetbrains.annotations.NotNull;

public class ObjJUnifiedMethodIndex extends ObjJMethodHeaderDeclarationsIndexBase<ObjJMethodHeaderDeclaration>{
    @NotNull
    public static final StubIndexKey<String, ObjJMethodHeaderDeclaration> KEY = IndexKeyUtil.createIndexKey(ObjJUnifiedMethodIndex.class);
    @NotNull
    private static final ObjJUnifiedMethodIndex INSTANCE = new ObjJUnifiedMethodIndex();
    private static final int VERSION = 1;

    private ObjJUnifiedMethodIndex() {
    }

    @Override
    public int getVersion() {
        return super.getVersion() + VERSION;
    }

    @NotNull
    @Override
    protected Class<ObjJMethodHeaderDeclaration> getIndexedElementClass() {
        return ObjJMethodHeaderDeclaration.class;
    }

    @NotNull
    public static ObjJUnifiedMethodIndex getInstance()
    {
        return INSTANCE;
    }

    @NotNull
    @Override
    public StubIndexKey<String, ObjJMethodHeaderDeclaration> getKey() {
        return KEY;
    }
}

File: src/org/cappuccino_project/ide/intellij/plugin/indices/ObjJUnifiedMethodIndex.kt
package org.cappuccino_project.ide.intellij.plugin.indices

import com.intellij.psi.stubs.StubIndexKey
import org.cappuccino_project.ide.intellij.plugin.psi.interfaces.ObjJMethodHeaderDeclaration

class ObjJUnifiedMethodIndex private constructor() : ObjJMethodHeaderDeclarationsIndexBase<ObjJMethodHeaderDeclaration<*>>() {

    protected override val indexedElementClass: Class<ObjJMethodHeaderDeclaration<*>>
        get() = ObjJMethodHeaderDeclaration::class.java

    override fun getVersion(): Int {
        return super.getVersion() + VERSION
    }

    override fun getKey(): StubIndexKey<String, ObjJMethodHeaderDeclaration<*>> {
        return KEY
    }

    companion object {
        val KEY: StubIndexKey<String, ObjJMethodHeaderDeclaration<*>> = IndexKeyUtil.createIndexKey<String, ObjJMethodHeaderDeclaration>(ObjJUnifiedMethodIndex::class.java)
        val instance = ObjJUnifiedMethodIndex()
        private val VERSION = 1
    }
}
